### PR TITLE
 first implementation of SQL Server Support

### PR DESCRIPTION
**Pull request by stmeyer**
can be extended for other RDMBS. Local Tests on SQL Server are passing. Hopefuly for MySQL will still pass, too 😄

### CODE REVIEW

File: src/Model/Table/QueuedJobsTable.php [hunk 1]
```diff
@@ -190,15 +204,25 @@ public function getStats() {
 	public function requestJob(array $capabilities, $group = null) {
 		$now = new Time();
 		$nowStr = $now->toDateTimeString();
+		$driverName = $this->_getDriverName();
 
 		$query = $this->find();
+		$age = $query->newExpr()->add('IFNULL(TIMESTAMPDIFF(SECOND, "' . $nowStr . '", notbefore), 0)');
+		switch ($driverName) {
+			case 'Postgres':
+			case 'Mysql':
```
Comment (dereuromark) [inline src/Model/Table/QueuedJobsTable.php]: Should we use class const here to avoid spelling mistakes?

Comment (dereuromark) [reply src/Model/Table/QueuedJobsTable.php]: Either you put $age down here, or you complete remove those case statements as they are currently no-ops.

Comment (stmeyer) [reply src/Model/Table/QueuedJobsTable.php]: Yes, I'll change tomorrow

**Comment by dereuromark** [comment]
Any ideas on tests in travis here etc?
Or should we just merge?

File: config/Migrations/20171019083500_fix_cloumn_length.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	 * More information on writing migrations is available here:
+	 * http://docs.phinx.org/en/latest/migrations.html#the-abstractmigration-class
+	 *
+	 * @return void
+	 */
+	public function change() {
+		$table = $this->table('queue_processes');
+
+		$table->changeColumn('pid', 'string', [
+		   'length' => 40,
```
Comment (dereuromark) [inline config/Migrations/20171019083500_fix_cloumn_length.php]: Why is this necessary?

Comment (stmeyer) [reply config/Migrations/20171019083500_fix_cloumn_length.php]: `pid` is set by https://github.com/dereuromark/cakephp-queue/blob/9759da22cc678fc6fc0516bc31774c0b662e298e/src/Model/Table/QueuedJobsTable.php#L479 PHP Doc of sha1 says

>If the optional raw_output is set to TRUE, then the sha1 digest is instead returned in raw binary format with a length of 20, otherwise the returned value is a 40-character hexadecimal number. 

Due to this we need 40 digits

**Comment by stmeyer** [comment]
I think testing sql server in travis is not possible. CakePHP uses appveyor to test on windows/with sql server.

In general tests would be gread but I never have used appveyor and I do not know how to set it up

**Comment by dereuromark** [comment]
Thank you!
Now only Postgres is not yet fully working :)
